pkgs/by-name/cw/cwtch-ui/package.nix

```
@@ -0,0 +1,60 @@
+{
+  cwtch,
+  fetchgit,
+  flutter,
+  lib,
+  tor,
+  zenity,
+}:
+let
+  runtimeBinDependencies = [
+    tor
+    zenity
+  ];
+in
+flutter.buildFlutterApplication rec {
+  pname = "cwtch-ui";
+  version = "1.15.1";
+  src = fetchgit {
+    url = "https://git.openprivacy.ca/cwtch.im/cwtch-ui";
+    rev = "v${version}";
+    hash = "sha256-+UtWhQMhm0UjY0kx3B5TwcBFhUfJma3rpeYls4XWy7I=";
+  };
+
+  # NOTE: The included pubspec.json does not exactly match the upstream
+  # pubspec.lock. With Flutter 3.24, a newer version of material_color_utilities
+  # is required than the upstream locked version. From a checkout of cwtch-ui
+  # 1.15.1, I ran `flutter pub upgrade material_color_utilities` on 2024-10-17.
+  # This upgraded material_color_utilities and its dependencies.
+  pubspecLock = lib.importJSON ./pubspec.json;
+  gitHashes = {
+    flutter_gherkin = "sha256-Y8tR84kkczQPBwh7cGhPFAAqrMZKRfGp/02huPaaQZg=";
+  };
+
+  flutterBuildFlags = [
+    "--dart-define"
+    "BUILD_VER=${version}"
+    "--dart-define"
+    "BUILD_DATE=1980-01-01-00:00"
+  ];
+
+  # These things are added to LD_LIBRARY_PATH, but not PATH
+  runtimeDependencies = [ cwtch ];
+
+  extraWrapProgramArgs = "--prefix PATH : ${lib.makeBinPath runtimeBinDependencies}";
+
+  postInstall = ''
+    mkdir -p $out/share/applications
+    sed "s|PREFIX|$out|" linux/cwtch.template.desktop >$out/share/applications/cwtch.desktop
+  '';
+
+  meta = {
+    description = "A decentralized, privacy-preserving, multi-party messaging app";
+    homepage = "https://cwtch.im/";
+    changelog = "https://cwtch.im/changelog/";
```

Wrong/dead link, and it should point to the changelog of the *current* version. The upstream anchors make it a bit annoying to sync it with the package version so maybe there should be a comment that says that the changelog needs to be manually updated

```suggestion
    changelog = "https://docs.cwtch.im/changelog/#cwtch-115---october-1st-2024";
```

---

The link is dead, so I've updated it to the new location.

However, the description of the `changelog` meta attribute ("A link or a list of links to the location of Changelog for a package. A link may use expansion to refer to the correct changelog version. Example: `"https://git.savannah.gnu.org/cgit/hello.git/plain/NEWS?h=v${version}"`") implies to me that pointing to the current version is not mandatory, and so I'm inclined to do it only if it's trivial.